sendto_silhouette.py

```diff
@@ -567,6 +584,24 @@ def effect(self):
 
         self.logEnvironment()
 
+        # Auto detect mark to document corner and mark to mark distance from Regmark layer (Quality of life improvement)
+        # Generated Regmarks from inkscape silhouette embeds ids so we can automatically derive offset calculations from it
+        doc_reg_x = doc_reg_y = doc_reg_width = doc_reg_length = 0
+        if self.svg.getElementById(REGMARK_TOP_LEFT_ID) is not None and self.svg.getElementById(REGMARK_TOP_RIGHT_ID) is not None and self.svg.getElementById(REGMARK_BOTTOM_LEFT_ID) is not None:
+            doc_reg_x = self.svg.unit_to_viewport(self.svg.getElementById(REGMARK_TOP_LEFT_ID).bounding_box(transform=True).x.minimum, "mm")
+            doc_reg_y = self.svg.unit_to_viewport(self.svg.getElementById(REGMARK_TOP_LEFT_ID).bounding_box(transform=True).y.minimum, "mm")
+            doc_reg_width = self.svg.unit_to_viewport(self.svg.getElementById(REGMARK_TOP_RIGHT_ID).bounding_box(transform=True).x.maximum, "mm") - doc_reg_x
+            doc_reg_length = self.svg.unit_to_viewport(self.svg.getElementById(REGMARK_BOTTOM_LEFT_ID).bounding_box(transform=True).y.maximum, "mm") - doc_reg_y
+            self.report(f"Detected Existing Registration Mark:: mark distance from document: Left={doc_reg_x}mm, Top={doc_reg_y}mm; mark to mark distance: X={doc_reg_width}mm, Y={doc_reg_length}mm;", 'log')
+
+        # Registration Mark Selection/Calcuation
+        # Note that any user defined values will overrride any auto calcuated values
+        reg_origin_X = self.options.regoriginx or doc_reg_x
+        reg_origin_Y = self.options.regoriginy or doc_reg_y
+        reg_width = self.options.regwidth or doc_reg_width or self.svg.to_dimensional(self.svg.viewport_width, "mm") - reg_origin_X * 2
+        reg_length = self.options.reglength or doc_reg_length or self.svg.to_dimensional(self.svg.viewport_height, "mm") - reg_origin_Y * 2
+        self.report(f"Using Registration Mark:: mark distance from document: Left={reg_origin_X}mm, Top={reg_origin_Y}mm; mark to mark distance: X={reg_width}mm, Y={reg_length}mm;", 'log')
+
```

Comment:
wow great job, just some last mini tweaks before merging...
- [ ] just realized we could replace 
  - [ ] `x.minimum` with `left`
  - [ ] y.minimum with top
  - [ ] x.maximum with right
  - [ ] y.maximum with bottom
- [ ] shift this whole regmark block to it's own function `def detect_regmark()` i.e. next to def initDocScale() and only call it here... this will help to keep the code more structured
  - [ ] this involves to rename i.e. `doc_reg_x` to `self.doc_reg_x`, etc. all over the places
:smile:

Reply:
All good. Fixed